spec/components/spotlight/analytics/dashboard_component_spec.rb

```diff
@@ -39,6 +57,66 @@
     expect(rendered).to have_content '123'
   end
 
+  it 'has valid start and end date params' do
+    start_date = '2024-01-01'
+    end_date = '2024-01-02'
+    vc_test_controller.params[:start_date] = start_date
+    vc_test_controller.params[:end_date] = end_date
+    expect(get_min_max('start_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(get_min_max('end_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(rendered).to have_field('start_date', with: start_date)
+    expect(rendered).to have_field('end_date', with: end_date)
+    expect(rendered).to have_content "User activity from #{component.formatted_date(start_date)} to #{component.formatted_date(end_date)}"
+  end
+
+  it 'has invalid start and end date params' do
+    vc_test_controller.params[:start_date] = '2024-41-01'
+    vc_test_controller.params[:end_date] = '2024-31-02'
+    expect(get_min_max('start_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(get_min_max('end_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(rendered).to have_field('start_date', with: default_start_date)
+    expect(rendered).to have_field('end_date', with: default_end_date)
+    expect(rendered).to have_content "User activity from #{component.formatted_date(default_start_date)} to #{component.formatted_date(default_end_date)}"
+  end
+
+  it 'has a start date larger than end date' do
+    vc_test_controller.params[:start_date] = '2024-02-02'
+    vc_test_controller.params[:end_date] = '2024-02-01'
+    expect(get_min_max('start_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(get_min_max('end_date')).to eq({ max: default_end_date, min: default_min_date })
+    expect(rendered).to have_field('start_date', with: default_start_date)
+    expect(rendered).to have_field('end_date', with: default_end_date)
+    expect(rendered).to have_content "User activity from #{component.formatted_date(default_start_date)} to #{component.formatted_date(default_end_date)}"
+  end
+
+  context 'Spotlight::Engine.config.ga_date_range is set' do
+    before do
+      Spotlight::Engine.config.ga_date_range = { 'start_date' => '2024-10-10', 'end_date' => '2024-12-10' }
+    end
+
+    after do
+      Spotlight::Engine.config.ga_date_range = { 'start_date' => nil, 'end_date' => nil }
+    end
+
+    it 'has start and end date selectors' do
+      expect(get_min_max('start_date')).to eq({ max: '2024-12-10', min: '2024-10-10' })
+      expect(get_min_max('end_date')).to eq({ max: '2024-12-10', min: '2024-10-10' })
+      expect(rendered).to have_field('start_date', with: default_start_date)
+      expect(rendered).to have_field('end_date', with: default_end_date)
+      expect(rendered).to have_content 'User activity over the past year'
+    end
+
+    it 'has invalid start and end date params' do
```

Comment:
This applies generally to most of the `it` statements here, but I think the tests would be clearer with context blocks that describe the condition followed by an it block that describes the expected behavior, (or just an it block that describes both the condition and expected behavior), such as:

```
context 'when invalid start and end dates are supplied ' do
  it 'displays an error and falls back to the default start and end dates'
```